src/votekit/metrics/distances.py

```diff
@@ -0,0 +1,108 @@
+from votekit.profile import PreferenceProfile
+from votekit.graphs.models import BallotGraph
+import numpy as np
+import ot  # type: ignore
+import networkx as nx  # type: ignore
+from typing import Union, Optional
+
+
+def earth_mover_dist(pp1: PreferenceProfile, pp2: PreferenceProfile) -> int:
+    """
+    Computes the earth mover distance between two elections. \n
+    Assumes both elections share the same candidates
+    """
+    # create ballot graph
+    graph = BallotGraph(source=pp2, complete=True)
+    ballot_graph = graph.from_profile(profile=pp2, complete=True)
+
+    # Solving Earth Mover Distance
+    electA_distr = np.array(em_array(pp=pp1))
+    electB_distr = np.array(em_array(pp=pp2))
+
+    # Floyd Warshall Shortest Distance alorithm. Returns a dictionary of shortest path for each node
+    fw_dist_dict = nx.floyd_warshall(ballot_graph)
+    keys_list = sorted(fw_dist_dict.keys())
+    cost_matrix = np.zeros((len(keys_list), len(keys_list)))
+    for i in range(len(keys_list)):
+        node_dict = fw_dist_dict[keys_list[i]]
+        cost_col = [value for key, value in sorted(node_dict.items())]
+        cost_matrix[i] = cost_col
+    earth_mover_matrix = ot.emd(electA_distr, electB_distr, cost_matrix)
+
+    # Hadamard Product = Earth mover dist between two matrices
+    earth_mover_dist = np.sum(np.multiply(cost_matrix, earth_mover_matrix))
+    return earth_mover_dist
+
+
+def lp_dist(
+    pp1: PreferenceProfile,
+    pp2: PreferenceProfile,
+    p_value: Optional[Union[int, str]] = 1,
+) -> int:
+    """
+    Computes the L_p distance between two election distributions.
+    Use 'inf' for infinity norm. \n
+    Assumes both elections share the same candidates.
+    """
+    election_arrays = profilePairs_to_arrays(pp1, pp2)
+    electA_distr = np.array(election_arrays[0])
+    electB_distr = np.array(election_arrays[1])
+
+    if isinstance(p_value, int):
+        sum = 0
+        for i in range(len(electA_distr)):
+            diff = (abs(electA_distr[i] - electB_distr[i])) ** p_value
+            sum += diff
+        lp_dist = sum ** (1 / p_value)
+        return lp_dist
+
+    elif p_value == "inf":
+        diff = [x - y for x, y in zip(electA_distr, electB_distr)]
+        return max(diff)
+
+    else:
+        raise ValueError("Unsupported input type")
+
+
+# helper functions
+# these functions comvert a list of preference profiles into distribution arrays
+def profilePairs_to_arrays(
```

Comment:
this is a super useful function!! but instead of limiting it to pairs, lets either allow users to input a list profiles or change it to profile_to_array() (i.e. single profile) and loop through when we use, bit more abstract that way!

Reply:
done